src/Fantomas.Tests/TypeDeclarationTests.fs

```diff
@@ -1345,3 +1345,18 @@ type Message =
     { [<JsonProperty("body")>]
       Body: string }
 """
+
+[<Test>]
+let ``type constraint on type definition, 887`` () =
+    formatSourceString false """
+type OuterType =
+    abstract Apply<'r>
+        : InnerType<'r>
+        -> 'r when 'r : comparison
+"""  { config with SpaceBeforeColon = true }
+    |> prepend newline
+    |> should equal """
+type OuterType =
+    abstract Apply<'r> : InnerType<'r>
+        -> 'r when 'r : comparison
```

Comment:
Would it be possible to get a test where there are two type constraints to make sure it gets them both

Reply:
Sure, could you maybe post a small example of that?
I'm less familiar with constraints, unfortunately.

Reply:
Actually, ignore this request. Another PR handles the issue that 'and' is missing!